src/java/org/apache/cassandra/service/accord/api/PartitionKey.java

```diff
@@ -147,8 +148,8 @@ public <V> int serialize(PartitionKey key, V dst, ValueAccessor<V> accessor, int
         public PartitionKey deserialize(DataInputPlus in, int version) throws IOException
         {
             TableId tableId = TableId.deserialize(in);
-            TableMetadata metadata = Schema.instance.getExistingTableMetadata(tableId);
-            DecoratedKey key = metadata.partitioner.decorateKey(ByteBufferUtil.readWithShortLength(in));
+            IPartitioner partitioner = Schema.instance.getExistingTablePartitioner(tableId);
```

Comment:
using this new method lets test focus only on the partitioner and not define a random table just to access a partitioner